src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonObjectFormatter.java

```diff
@@ -0,0 +1,363 @@
+import java.util.List;
+import java.util.Map;
+
+public class TestJacksonObjectFormatter {
+
+  private class TestParentParser implements IParser {
+
+    private final boolean isArray;
+
+    public TestParentParser( final boolean isArray ) {
+      this.isArray = isArray;
+    }
+
+    @Override
+    public PrimitiveObject get(final String key ) throws IOException {
+      return null;
+    }
+
+    @Override
+    public PrimitiveObject get( final int index ) throws IOException {
+      return null;
+    }
+
+    @Override
+    public IParser getParser( final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return new TestPrimitiveParser( isArray );
+      }
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public IParser getParser( final int index ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public String[] getAllKey() throws IOException {
+      return new String[]{ "key1" };
+    }
+
+    @Override
+    public boolean containsKey( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public int size() throws IOException {
+      return 0;
+    }
+
+    @Override
+    public boolean isArray() throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean isMap() throws IOException {
+      return true;
+    }
+
+    @Override
+    public boolean isStruct() throws IOException {
+      return true;
+    }
+
+    @Override
+    public boolean hasParser( final int index ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean hasParser( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public Object toJavaObject() throws IOException {
+      return null;
+    }
+
+  }
+
+  private class TestPrimitiveParser implements IParser {
+
+    private final boolean isArray;
+
+    public TestPrimitiveParser( final boolean isArray ) {
+      this.isArray = isArray;
+    }
+
+    @Override
+    public PrimitiveObject get(final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return new StringObj( "a" );
+      } else if ( "key2".equals( key ) ) {
+        return new StringObj( "b" );
+      } else if ( "key3".equals( key ) ) {
+        return new StringObj( "c" );
+      }
+      return null;
+    }
+
+    @Override
+    public PrimitiveObject get( final int index ) throws IOException {
+      if ( index == 0 ) {
+        return new StringObj( "a" );
+      } else if ( index == 1 ) {
+        return new StringObj( "b" );
+      } else if ( index == 2 ) {
+        return new StringObj( "c" );
+      }
+      return null;
+    }
+
+    @Override
+    public IParser getParser( final String key ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public IParser getParser( final int index ) throws IOException {
+      return new JacksonNullParser();
```

Comment:
Since get(String key) returns a value when the key is "key1", "key2" or "key3" and get(int index) return a value when the index between 0 and 2, but getParser(Sting key) and getParser(int index) return JacksonNullParser. It feels to me strange.

Reply:
This parser is a parser for unnested data.
Therefore, this parser does not have a child parser.

Reply:
Thank you